src/repository/documents.rs

```diff
@@ -110,8 +120,31 @@ impl DocumentRepository for PgDocumentRepository {
             Ok(_result) => Ok(()),
             Err(err) => {
                 error!(%err);
-                return Err(DocumentInsertError::Unknown);
+                Err(DocumentInsertError::Unknown)
             }
         }
     }
+
+    async fn read_file(&self, document_id: i32) -> Result<PathBuf, DocumentGetError> {
+        // We don't check any access privileges for now
+        let file_path = get_document_path(document_id);
+
+        if !file_path.exists() {
+            if let Err(err) = fs::write(&file_path, "").await {
+                error!(%err);
+                return Err(DocumentGetError::Unknown);
+            }
+        }
```

Comment:
This is here since we don't create the file in the insert. I quickly made it like this, but we could move it to the insert and remove it from both read and write. The problem with that is that it might lead to inconsistent states if for example the database transaction succeedes and file creation fails

Reply:
That would be much better. About the creation itself, I see some possible solutions:
1. Create file first, then add record to db, and if it fails, remove the file
2. Utilize db transactions - start transaction, add record, create file, commit transaction
Generally, we won't create too many files, but if there is a reference in database, then the file should exist

Reply:
both 1. and 2. have the same problem i described. 1.: Create file succeeds, add record fails, remove file fails. 2.: We already use transactions and is exactly what I described.